### PR TITLE
B2c cart endpoints

### PR DESCRIPTION
**Pull request by pyvovarov-s**
B2c cart endpoints

### CODE REVIEW

File: tests/api/b2c/glue/cart_endpoints/items/positive.robot [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    And Response body parameter should be greater than:    [data][attributes][totals][priceToPay]    1
+    And Response body parameter should not be EMPTY:    [data][attributes][discounts]
+    And Response body parameter should not be EMPTY:    [data][links][self]
+
+Add_two_items_to_cart_with_included_items_concrete_products_and_abstract_products
+   [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+          ...  AND    I set Headers:    Authorization=${token}
+          ...  AND    Find or create customer cart
+          ...  AND    Cleanup all items in the cart:    ${cart_id}
+    When I send a POST request:    /carts/${cart_id}/items?include=items,concrete-products,abstract-products    {"data": {"type": "items","attributes": {"sku": "${add_cart_item_sku}", "quantity": "1"}}}
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/cart_endpoints/items/positive.robot:35]: why does test say it adds 2 items but in fact adds only 1?

File: resources/environments/environments_api_b2c.json [hunk 2]
```diff
@@ -14,11 +14,11 @@
...
 
-        "yves_second_user_email": "bill.martin@spryker.com",
-        "yves_second_user_first_name": "Martin",
-        "yves_second_user_last_name": "Bill",
+        "yves_second_user_email": "spencor.hopkin@spryker.com",
+        "yves_second_user_first_name": "Hopkin",
+        "yves_second_user_last_name": "Spencor",
         "yves_second_user_password": "change123",
-        "yves_second_user_reference": "DE--4",
+        "yves_second_user_reference": "DE--1",
```
[the remark this answers is not in the document]
Comment (pyvovarov-s) [reply resources/environments/environments_api_b2c.json:21]: reverted

File: tests/api/b2b/glue/cart_endpoints/carts/positive.robot [hunk 3]
```diff
@@ -309,4 +309,4 @@ Delete_cart_by_cart_id
     And Response reason should be:    No Content
     When I send a GET request:    /carts/${cart_id}
     Then Response status code should be:    404
-    And Array in response should contain property with value:    [errors]    detail    Cart with given uuid not found.
\ No newline at end of file
+    And Array in response should contain property with value:    [errors]    detail    Cart with given uuid not found.
```
[the remark this answers is not in the document]
Comment (pyvovarov-s) [reply tests/api/b2b/glue/cart_endpoints/carts/positive.robot:312]: the end of file was corrected here, file should be finished with empty line

File: tests/api/b2c/glue/cart_endpoints/carts/negative.robot [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+    And Response reason should be:    Precondition Failed
+    And Response should return error message:    If-Match header value is invalid.
+    And Response should return error code:    006
+
+Update_cart_without_cart_id
+      [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+        ...  AND    I set Headers:    Authorization=${token}
+        ...  AND    Find or create customer cart
+        ...  AND    Save Header value to a variable:    ETag    header_tag
+        ...  AND    I set Headers:    Authorization=${token}    If-Match=${header_tag}
```
[the remark this answers is not in the document]
Comment (pyvovarov-s) [reply tests/api/b2c/glue/cart_endpoints/carts/negative.robot:168]: we need it or we will get either  invalid token  or wrong ETag

File: tests/api/b2c/glue/cart_endpoints/carts/negative.robot [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+    And Response should return error message:    Unauthorized request.
+
+Create_cart_with_invalid_access_token
+    When I send a POST request:    /carts    {"data": {"type": "carts","attributes": {"priceMode": "${gross_mode}","currency": "${currency_code_eur}","store": "${store_de}","name": "${test_cart_name}"}}}
+    Then Response status code should be:    403
+    And Response reason should be:    Forbidden
+    And Response should return error message:    Missing access token.
+    And Response should return error code:    002
+
+Create_cart_when_cart_already_exist
```
[the remark this answers is not in the document]
Comment (pyvovarov-s) [reply tests/api/b2c/glue/cart_endpoints/carts/negative.robot:117]: cause in b2c it's not possible  to remove the card

File: resources/common/common_api.robot [hunk 6]
```diff
@@ -1180,3 +1180,39 @@ Cleanup existing customer addresses:
...
+    [Documentation]    This keyword creates or retrieves cart for the current customer token.
+        ...
+        ...     This keyword does not accept any arguments. Makes GET request in order to fetch cart for the customer or creates it otherwise.
+        ...
+        ${response}=    I send a GET request:    /carts
+        Save value to a variable:    [data][0][id]    cart_id
+        ${hasCart}    Run Keyword and return status     Should not be empty    ${cart_id}
+        Log    cart_id:${cart_id}
+        Run Keyword Unless    ${hasCart}    I send a POST request:    /carts    {"data": {"type": "carts","attributes": {"priceMode": "${gross_mode}","currency": "${currency_code_eur}","store": "${store_de}"}}}
+        Run Keyword Unless    ${hasCart}    Save value to a variable:    [data][id]    cart_id
```
[the remark this answers is not in the document]
Comment (pyvovarov-s) [reply resources/common/common_api.robot:1194]: Documentation is Fixed but I suggest to leave  variable creation inside keyword. There is no any chance to use such keyword without variable creation and in order to have it in abstract I wold go in this way

**Review by OKrivtsova — APPROVED** [review]
run b2b tests to confirm changes  did not break tests.
merged fixes

**Pull request merged**